tests/test_execution_compute_plan.py

```diff
@@ -294,3 +294,35 @@ def test_compute_plan_circular_dependency_failure(global_execution_env):
         session.add_compute_plan(cp_spec)
 
     assert 'missing dependency among inModels IDs' in str(e.value)
+
+
+def test_execution_compute_plan_cancelled(global_execution_env):
+    factory, network = global_execution_env
+    session = network.sessions[0].copy()
+
+    dataset = session.state.datasets[0]
+
+    spec = factory.create_algo()
+    algo = session.add_algo(spec)
+
+    cp_spec = factory.create_compute_plan()
+    previous_traintuple = None
+    for data_sample_key in dataset.train_data_sample_keys:
+        previous_traintuple = cp_spec.add_traintuple(
+            algo=algo,
+            dataset=dataset,
+            data_samples=[data_sample_key],
+            in_models=[previous_traintuple] if previous_traintuple else None
+        )
+
+    cp = session.add_compute_plan(cp_spec)
+    first_traintuple = [t for t in cp.attach(session).list_traintuple() if t.rank == 0][0]
+    first_traintuple.future().wait()
+    cp = session.cancel_compute_plan(cp.compute_plan_id)
+
+    assert cp.status == assets.TupleStatus.canceled
```

Comment:
minor: as said in a previous comment, we should not compare a compute plan status with a tuple status. 

I see 2 options:
- have a `ComputePlanStatus` class
- rename `TupleStatus` to a more generic name: `FutureStatus` ?

Reply:
I like having a single class with a generic name but FutureStatus doesn't seem right. I really didn't see any issue with using TupleStatus though, especially since the compute plan status is inferred from its tuples' statuses. I'll create a ComputePlanStatus so that we can move forward.

Reply:
I agree, `FutureStatus` does not seem right. We could directly use `Status`, we don't have other statuses.

Reply:
👍